dropwizard/src/test/java/com/linecorp/armeria/server/dropwizard/ArmeriaConnectorFactoryTest.java

```diff
@@ -0,0 +1,280 @@
+        }
+    }
+
+    @Nested
+    class ArmeriaHttpsConnectorFactoryTest {
+        private final YamlConfigurationFactory<ArmeriaHttpsConnectorFactory> configFactory =
+                new YamlConfigurationFactory<>(
+                        ArmeriaHttpsConnectorFactory.class, validator, objectMapper, "dw");
+
+        @Test
+        public void withoutKeyStore_shouldNotValidate() throws Exception {
+            final File yml = new File(resourceFilePath("yaml/connector/https-minimal.yaml"));
+            verifyHttpsConnectorWithoutKeyStoreIsInvalid(configFactory, yml);
+        }
+
+        @Test
+        public void withKeyStore_shouldBuild() throws Exception {
+            final File yml = new File(resourceFilePath("yaml/connector/https-keystore.yaml"));
+            validateHttpsConnectorWithKeyStore(configFactory, yml);
+        }
+
+        @Test
+        public void testFactoryDefaults() {
+            final String f1 = ResourceHelpers.resourceFilePath("f.crt");
+            final String f2 = ResourceHelpers.resourceFilePath("f.jks");
+
+            final ArmeriaHttpsConnectorFactory factory =
+                    (ArmeriaHttpsConnectorFactory) ArmeriaHttpsConnectorFactory.build(f1, f2, null);
+            assertThat(factory.getPort()).isEqualTo(8082);
+            assertThat(factory.getSessionProtocols()).containsOnly(SessionProtocol.HTTPS);
+            assertThat(factory.getKeyCertChainFile()).isEqualTo(f1);
+            assertThat(factory.getKeyStorePath()).isEqualTo(f2);
+            assertThat(factory.isSelfSigned()).isFalse();
+        }
+    }
+
+    @Nested
+    class ArmeriaH2ConnectorFactoryTest {
+        private final YamlConfigurationFactory<ArmeriaH2ConnectorFactory> configFactory =
+                new YamlConfigurationFactory<>(
+                        ArmeriaH2ConnectorFactory.class, validator, objectMapper, "dw");
+
+        @Test
+        public void withoutKeyStore_shouldNotValidate() throws Exception {
+            final File yml = new File(resourceFilePath("yaml/connector/h2-minimal.yaml"));
+            verifyHttpsConnectorWithoutKeyStoreIsInvalid(configFactory, yml);
+        }
+
+        @Test
+        public void shouldBuild() throws Exception {
+            final File yml = new File(resourceFilePath("yaml/connector/h2-keystore.yaml"));
+            final ConnectorFactory factory = configFactory.build(yml);
+            assertThat(factory)
+                    .isInstanceOf(Discoverable.class)
+                    .isInstanceOf(ArmeriaH2ConnectorFactory.class);
+        }
+
+        @Test
+        public void testFactoryDefaults() throws CertificateException, SSLException {
+            final String f1 = ResourceHelpers.resourceFilePath("f.crt");
+            final String f2 = ResourceHelpers.resourceFilePath("f.jks");
+            final ArmeriaH2ConnectorFactory factory =
+                    (ArmeriaH2ConnectorFactory) ArmeriaH2ConnectorFactory.build(f1, f2, null);
```

Comment:
Is there a way to generate legitimate JKS certificates for testing?

Reply:
So far we used `SelfSignedCertificateRule/Extension` but it doesn’t generate JKS but PEM.. I think it’s fine to use a pre-generated resource like you did.

Reply:
I just touched empty files. Wasn't sure if there was a better way to make valid files.